### PR TITLE
Backend/review pmr restructure

### PR DESCRIPTION
**Pull request by rachitkh4**

### CODE REVIEW

File: backend/core/core/apis/schemas/requests/ai_request.py [hunk 1]
```diff
@@ -100,9 +100,10 @@ class AudioUpload(BaseModel):
 
 class AddFeedback(BaseModel):
     pmr_id: str
-    feedback_response: str
-    feeback_reason: Optional[str]
-    feedback_comment: Optional[str]
+    transcript_timestamp: str
+    summary_feedback: str
```
Comment (arpitkjain7) [inline backend/core/core/apis/schemas/requests/ai_request.py:104]: add generation feedback items

File: backend/core/core/controllers/ai_controller.py [hunk 2]
```diff
@@ -176,17 +184,18 @@ def add_feedback(self, request):
...
-            summary_feedback = request_dict.get("feedback_response")
-            feedback_details = {
-                "feeback_reason": request_dict.get("feeback_reason"),
-                "feedback_comment": request_dict.get("feedback_comment"),
-            }
             db_request = {
-                "id": pmr_id,
-                "summary_feedback": summary_feedback,
-                "feedback_details": feedback_details,
+                "pmr_id": pmr_id,
```
Comment (arpitkjain7) [inline backend/core/core/controllers/ai_controller.py:188]: add generation feedback values

File: backend/core/core/controllers/pmr_controller.py [hunk 3]
```diff
@@ -1155,6 +1180,12 @@ def submit_pmr_v1(self, pmr_request, appointment_request):
                         "notes": pmr_request.notes,
                     }
                 )
+                self.CRUDConsultationNotes.update_by_pmrId(
```
Comment (arpitkjain7) [inline backend/core/core/controllers/pmr_controller.py:1183]: add log messages

File: backend/core/core/controllers/pmr_controller.py [hunk 4]
```diff
@@ -198,14 +212,25 @@ def create_pmr_v2(self, request):
             )
             # TODO: Optimise the function to reduce additional query
             pmr_details = self.CRUDPatientMedicalRecord.read_joined(pmr_id=pmr_id)
-            del pmr_details["raw_transcripts"]
+            consultation_notes_list = pmr_details.pop("consultation_notes_list")
```
Comment (arpitkjain7) [inline backend/core/core/controllers/pmr_controller.py:215]: add condition to check is_utilized flag

File: backend/core/core/controllers/pmr_controller.py [hunk 5]
```diff
@@ -83,7 +87,16 @@ def get_pmr(self, pmr_id: str):
             logging.info("executing get pmr function")
             logging.info(f"Getting the PMR record for {pmr_id=}")
             pmr_details = self.CRUDPatientMedicalRecord.read_joined(pmr_id=pmr_id)
-            del pmr_details["raw_transcripts"]
+            consultation_notes_list = pmr_details.pop("consultation_notes_list")
+            consolidated_summaries = {}
+            for current_dict in consultation_notes_list:
+                consolidated_summaries = consolidate_summaries(
```
Comment (arpitkjain7) [inline backend/core/core/controllers/pmr_controller.py:93]: add condition to check is_utilized flag

File: backend/core/core/crud/hims_consultation_notes_crud.py [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+from core.orm_models.hims_consulationNotes import ConsultationNotes
+from datetime import datetime
+from pytz import timezone
+
+logging = logger(__name__)
+
+
+class CRUDConsultationNotes:
+    def create(self, **kwargs):
+        """[CRUD function to create a new User record]
```
Comment (arpitkjain7) [inline backend/core/core/crud/hims_consultation_notes_crud.py:11]: correct log messages

File: backend/core/core/crud/hims_patientMedicalRecord_crud.py [hunk 7]
```diff
@@ -500,6 +501,16 @@ def read_joined(self, pmr_id: str):
                 pmr_obj = transaction_session.query(PatientMedicalRecord).get(pmr_id)
                 pmr_dict = pmr_obj.__dict__
 
+                consultation_notes_result = (
+                    transaction_session.query(ConsultationNotes)
+                    .filter(ConsultationNotes.pmr_id == pmr_id)
+                    .filter(ConsultationNotes.is_utilized == False)
```
Comment (arpitkjain7) [inline backend/core/core/crud/hims_patientMedicalRecord_crud.py:507]: remove this filter from CRUD

File: backend/core/core/orm_models/hims_consulationNotes.py [hunk 8]
```diff
@@ -0,0 +1,22 @@
+from sqlalchemy import Column, String, JSON, DateTime, ForeignKey, Boolean
+from core import Base
+
+
+class ConsultationNotes(Base):
+    __tablename__ = "consultationNotes"
+    __table_args__ = {"extend_existing": True}
+    timestamp = Column(String, primary_key=True)
```
Comment (arpitkjain7) [inline backend/core/core/orm_models/hims_consulationNotes.py:8]: add unique key constraint

**Review by arpitkjain7 — CHANGES_REQUESTED** [review]

File: backend/core/core/controllers/pmr_controller.py (hunk 5 above)
Comment (arpitkjain7) [reply backend/core/core/controllers/pmr_controller.py:93]: Done

File: backend/core/core/crud/hims_patientMedicalRecord_crud.py (hunk 7 above)
Comment (arpitkjain7) [reply backend/core/core/crud/hims_patientMedicalRecord_crud.py:507]: Done

File: backend/core/core/apis/schemas/requests/ai_request.py [hunk 9]
```diff
@@ -82,7 +82,7 @@ class AdditionalNotes(BaseModel):
 
 
 class MedicalSummary(BaseModel):
-    pmr_id: str
```
[the remark this answers is not in the document]
Comment (rachitkh4) [reply backend/core/core/apis/schemas/requests/ai_request.py]: fixed

File: backend/core/core/apis/schemas/requests/ai_request.py (hunk 1 above)
Comment (rachitkh4) [reply backend/core/core/apis/schemas/requests/ai_request.py:104]: added

File: backend/core/core/controllers/ai_controller.py (hunk 2 above)
Comment (rachitkh4) [reply backend/core/core/controllers/ai_controller.py:188]: done

File: backend/core/core/orm_models/hims_consulationNotes.py (hunk 8 above)
Comment (arpitkjain7) [reply backend/core/core/orm_models/hims_consulationNotes.py:8]: Done

**Review by arpitkjain7 — APPROVED** [review]

**Pull request merged**